File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, arr: List[int]) -> int:
        if not arr:
            return 0
        if len(arr)==1:
            if arr[0]==1 :
                return 1 
            else:
                return 0
        maxi=0
        curr_c=0
        for ele in arr:
            if ele==1:
                curr_c+=1
            else:
                curr_c=0
            maxi=max(curr_c,maxi)
        return maxi


















        # if len(arr)==0:
        #     return 0
        # if len(arr)==1:
        #     if arr[0]==1:
        #         return 1
        #     else:
        #         return 0
        
        # max_c=0
        # curr_c=0
        
        # for ele in arr:
        #     if ele==1 :
        #         curr_c+=1
        #     else:
        #         curr_c=0
        #     max_c=max(max_c, curr_c)
            
        # return max_c

File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] arr) {
        int maxi_c=0;
        int curr_c=0;
        for (int i=0;i<arr.length;i++){
            if (arr[i]==1){
                curr_c+=1;
            }
            else{
                curr_c=0;
            }
            maxi_c=Math.max(curr_c,maxi_c);
        }
        return maxi_c;











        // if (arr==null || arr.length==0){
        //     return 0;
        // }
        // int maxi=0;
        // int curr_c=0;
        // int n=arr.length;
        // for (int i=0; i<n;i++){
        //     if (arr[i]==1){
        //         curr_c+=1;
        //     }
        //     else {
        //         curr_c=0;
        //     }
        //     maxi=Math.max(maxi, curr_c);

        // }
        // return maxi;
    }
}